appender.go

```diff
@@ -42,6 +42,45 @@ func TestAppender(t *testing.T, store types.Storager) {
 			})
 		})
 
+		Convey("When CreateAppend with an existing appendable object", func() {
+			path := uuid.NewString()
+			o, err := ap.CreateAppend(path)
+
+			defer func() {
+				err := store.Delete(path)
+				if err != nil {
+					t.Error(err)
+				}
+			}()
+
+			Convey("The first returning error should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+
+			size := rand.Int63n(4 * 1024 * 1024) // Max file size is 4MB
+			r := io.LimitReader(randbytes.NewRand(), size)
+
+			_, err = ap.WriteAppend(o, r, size)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			o, err = ap.CreateAppend(path)
+
+			Convey("The second returning error also should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+
+			Convey("The Object Mode should be appendable", func() {
+				// Append object's mode must be appendable.
+				So(o.Mode.IsAppend(), ShouldBeTrue)
+			})
+
+			Convey("The object append offset should be 0", func() {
+				So(o.MustGetAppendOffset(), ShouldBeZeroValue)
```

Comment:
Is append offset the same as size?

Reply:
Yes, the next append offset should match the current object length. 
Maybe I need to add the size(content-length) check.

Reply:
Does this mean that append offset is redundant?

Reply:
I mean we can retain the offset check and add the size check when `CreateAppend` with an existing appendable object.

Reply:
I know what you mean here. Of course yes. 

But I was wondering whether we can remove the field if it's not needed.

Reply:
Some services require `CommitAppend` to finish an append process, and until it was committed, the object will not have valid `content-length`. That's why we introduced [GSP-44: Add CommitAppend in Appender](https://github.com/beyondstorage/go-storage/blob/master/docs/rfcs/44-commit-append.md).

Reply:
got it